core/templates/components/question-directives/question-player/question-player.component.ts

```diff
@@ -563,6 +556,13 @@ export class QuestionPlayerComponent implements OnInit, OnDestroy {
             QuestionPlayerConstants.HASH_PARAM.length));
 
         if (resultHashString) {
+          this.siteAnalyticsService.registerPracticeSessionEndEvent(
+            this.urlService.getClassroomUrlFragmentFromUrl(),
+            this.urlService.getTopicUrlFragmentFromLearnerUrl(),
+            Object.keys(this.scorePerSkillMapping).toString(),
+            this.totalQuestions,
```

Comment:
In the deleted lines of this file, when `registerPracticeSessionEndEvent()` is called, a local variable is used here instead (i.e. when `calculateScores()` is called, a local variable `totalQuestions` is first set to the length of `questionStateData`, and then used in the `registerPracticeSessionEndEvent()` function call).

Here in the added lines, the class property is directly used instead. Will its value be defined when the function call is made? (I don't see any `updateTotalQuestions()` calls before this in the init hook -- only once, inside a handler function for a subscription).

Reply:
Yup, since this method is only executed at the end of a practice session, the total questions should remain set as the initial value. However, I noticed that we need the updated score here, so I moved `registerPracticeSessionEndEvent` to after `calculateScores`. Thanks for bringing this up Nithin!